src/LiveComponent/src/ComponentWithMultiStepFormTrait.php

```diff
@@ -0,0 +1,315 @@
+
+        foreach ($this->stepNames as $stepName) {
+            if ($this->currentStepName === $stepName) {
+                $found = true;
+
+                continue;
+            }
+
+            if ($found) {
+                $next = $stepName;
+
+                break;
+            }
+        }
+
+        if (null === $next) {
+            throw new \RuntimeException('No next forms available.');
+        }
+
+        $this->currentStepName = $next;
+        $this->getStorage()->persist(\sprintf('%s_current_step_name', self::prefix()), $this->currentStepName);
+
+        // If we have a next step, we need to resinstantiate the form and reset the form view and values.
+        $this->form = $this->instantiateForm();
+        $this->formView = null;
+
+        $formData = $this->getStorage()->get(\sprintf('%s_form_values_%s', self::prefix(), $this->currentStepName));
+
+        $this->formValues = [] === $formData
+            ? $this->extractFormValues($this->getFormView())
+            : $formData;
+
+        $this->form->setData($formData);
+    }
+
+    /**
+     * Moves to the previous step in the form.
+     *
+     * Retrieves the previous step's data and updates the form state.
+     * Throws a RuntimeException if no previous step is available.
+     */
+    #[LiveAction]
+    public function previous(): void
+    {
+        $found = false;
+        $previous = null;
+
+        foreach (array_reverse($this->stepNames) as $stepName) {
+            if ($this->currentStepName === $stepName) {
+                $found = true;
+
+                continue;
+            }
+
+            if ($found) {
+                $previous = $stepName;
+
+                break;
+            }
+        }
+
+        if (null === $previous) {
+            throw new \RuntimeException('No previous forms available.');
+        }
+
+        $this->currentStepName = $previous;
+        $this->getStorage()->persist(\sprintf('%s_current_step_name', self::prefix()), $this->currentStepName);
+
+        $this->form = $this->instantiateForm();
+        $this->formView = null;
+
+        $formData = $this->getStorage()->get(\sprintf(
+            '%s_form_values_%s',
+            self::prefix(),
+            $this->currentStepName,
+        ));
+
+        $this->formValues = $formData;
+        $this->form->setData($formData);
+    }
+
+    /**
+     * Checks if the current step is the first step.
+     *
+     * @return bool true if the current step is the first; false otherwise
+     */
+    #[ExposeInTemplate]
+    public function isFirst(): bool
+    {
+        return $this->currentStepName === $this->stepNames[array_key_first($this->stepNames)];
+    }
+
+    /**
+     * Checks if the current step is the last step.
+     *
+     * @return bool true if the current step is the last; false otherwise
+     */
+    #[ExposeInTemplate]
+    public function isLast(): bool
+    {
+        return $this->currentStepName === $this->stepNames[array_key_last($this->stepNames)];
```

Comment:
```suggestion
        return $this->currentStep === end($this->steps);
```

Reply:
i would like to use the new php functions array_key_first and array_key_last.

Reply:
haha the "new" :) 

The "problem" (a detail, to be honest) here is that you first look for a key, then get the value indexed by that key, then compare currentStepName is identical.

But we can update this in time if necessary :)